server/src/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java

```diff
@@ -0,0 +1,532 @@
+        final OutOfBandManagementDriver driver = getDriver(outOfBandManagementConfig);
+
+        Long actionTimeOut = timeout;
+        if (actionTimeOut == null) {
+            actionTimeOut = OutOfBandManagementActionTimeout.valueIn(host.getClusterId());
+        }
+
+        final OutOfBandManagementDriverPowerCommand cmd = new OutOfBandManagementDriverPowerCommand(options, actionTimeOut, powerOperation);
+        final OutOfBandManagementDriverResponse driverResponse = driver.execute(cmd);
+
+        if (driverResponse == null) {
+            throw new CloudRuntimeException(String.format("Out-of-band Management action (%s) on host (%s) failed due to no response from the driver", powerOperation, host.getUuid()));
+        }
+
+        if (powerOperation.equals(OutOfBandManagement.PowerOperation.STATUS)) {
+            transitionPowerState(driverResponse.toEvent(), outOfBandManagementConfig);
+        }
+
+        if (!driverResponse.isSuccess()) {
+            String errorMessage = String.format("Out-of-band Management action (%s) on host (%s) failed with error: %s", powerOperation, host.getUuid(), driverResponse.getError());
+            if (driverResponse.hasAuthFailure()) {
+                errorMessage = String.format("Out-of-band Management action (%s) on host (%s) failed due to authentication error: %s. Please check configured credentials.", powerOperation, host.getUuid(), driverResponse.getError());
+                sendAuthError(host, errorMessage);
+            }
+            if (!powerOperation.equals(OutOfBandManagement.PowerOperation.STATUS)) {
+                LOG.debug(errorMessage);
+            }
+            throw new CloudRuntimeException(errorMessage);
+        }
+
+        final OutOfBandManagementResponse response = new OutOfBandManagementResponse(outOfBandManagementDao.findByHost(host.getId()));
+        response.setSuccess(driverResponse.isSuccess());
+        response.setResultDescription(driverResponse.getResult());
+        response.setId(host.getUuid());
+        response.setOutOfBandManagementAction(powerOperation.toString());
+        return response;
+    }
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_HOST_OUTOFBAND_MANAGEMENT_CHANGE_PASSWORD, eventDescription = "updating out-of-band management password")
+    public OutOfBandManagementResponse changeOutOfBandManagementPassword(final Host host, final String newPassword) {
+        checkOutOfBandManagementEnabledByZoneClusterHost(host);
+        if (Strings.isNullOrEmpty(newPassword)) {
+            throw new CloudRuntimeException(String.format("Cannot change out-of-band management password as provided new-password is null or empty for the host %s.", host.getUuid()));
+        }
+        GlobalLock outOfBandManagementHostLock = GlobalLock.getInternLock(getOutOfBandManagementHostLock(host.getId()));
+        try {
+            if (outOfBandManagementHostLock.lock(ACQUIRE_GLOBAL_LOCK_TIMEOUT_FOR_HOST)) {
```

Comment:
The nesting in this method is very deep and hard to follow.  Please change this line to the following:

```
if (!outOfBandManagementHostLock.lock(ACQUIRE_GLOBAL_LOCK_TIMEOUT_FOR_HOST)) {
                LOG.error("Unable to acquire synchronization lock to change out-of-band management password for host id: " + host.getId());
                throw new CloudRuntimeException(String.format("Unable to acquire lock to change out-of-band management password for host (%s), please try after some time.", host.getUuid()));
}
```

This will eliminate the need for the ``else`` block and allow subsequent code to remove from the scope of the ``if`` block.